pkg/broker/ingress/handler_test.go

```diff
@@ -1,72 +1,143 @@
 package ingress
 
 import (
+	"bytes"
 	"context"
+	"encoding/json"
+	nethttp "net/http"
+	"strconv"
 	"testing"
 
 	cloudevents "github.com/cloudevents/sdk-go/v2"
-	"github.com/cloudevents/sdk-go/v2/client/test"
-	"github.com/google/go-cmp/cmp"
+	"github.com/cloudevents/sdk-go/v2/binding"
+	"github.com/cloudevents/sdk-go/v2/protocol/http"
+	"github.com/google/knative-gcp/pkg/broker/config"
+	"github.com/google/knative-gcp/pkg/broker/config/memory"
 )
 
-// TestHandlerChannelInbound uses a channel based mock client for both inbound and decouple.
-func TestHandlerChannelInbound(t *testing.T) {
-	inbound, ic := test.NewMockReceiverClient(t, 1)
-	decouple, dc := test.NewMockSenderClient(t, 1)
-	_, cleanup := createAndStartIngress(t, inbound, decouple)
-	defer cleanup()
+var topic = "topic1"
 
-	input := createTestEvent()
-	// Send an event to the inbound receiver client.
-	ic <- input
-	// Retrieve the event from the decouple sink.
-	output := <-dc
+var brokerConfig = &config.TargetsConfig{
+	Brokers: map[string]*config.Broker{
+		"ns1/broker1": {
+			Id:        "b-uid-1",
+			Name:      "broker1",
+			Namespace: "ns1",
+			DecoupleQueue: &config.Queue{
+				Topic: topic,
+			},
+		},
+	},
+}
 
-	if dif := cmp.Diff(input, output); dif != "" {
-		t.Errorf("Output event doesn't match input, dif: %v", dif)
-	}
+type testCase struct {
+	name    string
+	options []HandlerOption
+	// If port is different than the default(8080), a matching WithPort option should be provided.
+	port int
+	// in happy case, path should match the /<ns>/<broker> in the brokerConfig.
+	path  string
+	event *cloudevents.Event
+	// If method is empty, POST will be used as default.
+	method string
+	// body and header can be specified if the client is making raw HTTP request instead of via cloudevents.
+	body     map[string]string
+	header   nethttp.Header
+	wantCode int
 }
 
-// TestHandlerDefaultHTTPInbound uses default HTTP inbound client and a channel based mock decouple
-// sink.
-func TestHandlerDefaultHTTPInbound(t *testing.T) {
-	decouple, dc := test.NewMockSenderClient(t, 1)
-	_, cleanup := createAndStartIngress(t, nil, decouple)
-	defer cleanup()
+func TestHandler(t *testing.T) {
+	tests := []testCase{
```

Comment:
I recommend inling the `testCase` struct here, rather than defining it outside the function (unless you intend to reuse it somewhere else):

```suggestion
	tests := []struct{
		name    string
		options []HandlerOption
		// If port is different than the default(8080), a matching WithPort option should be provided.
		port int
		// in happy case, path should match the /<ns>/<broker> in the brokerConfig.
		path  string
		event *cloudevents.Event
		// If method is empty, POST will be used as default.
		method string
		// body and header can be specified if the client is making raw HTTP request instead of via cloudevents.
		body     map[string]string
		header   nethttp.Header
		wantCode int
	}{
```

Reply:
Yes the `createRequest` helper function needs it.